packages/cells/src/widget.ts

```diff
@@ -643,6 +616,28 @@ namespace CodeCell {
     rendermime: RenderMime;
   }
 
+  /**
+   * Execute a cell given a client session.
+   */
+  export
+  function execute(cell: CodeCell, session: IClientSession): Promise<KernelMessage.IExecuteReplyMsg> {
+    let model = cell.model;
+    let code = model.value.text;
+    if (!code.trim() || !session.kernel) {
+      model.executionCount = null;
+      model.outputs.clear();
+      return Promise.resolve(void 0);
+    }
+
+    model.executionCount = null;
+    cell.setPrompt('*');
+    model.trusted = true;
+
+    return OutputArea.execute(cell.outputArea, code, session).then(msg => {
```

Comment:
I have a very slight preference to swap the order of the `code` and `outputArea` arguments as it more closely matches how I think about the logic = "run this code and send the output to this output area."